concepts/foreach-loops/introduction.md

```diff
@@ -1,54 +1,56 @@
-# Introduction
+# Introduction to For-Each Loops
 
-TODO: the content below is copied from the exercise introduction and probably needs rewriting to a proper concept introduction
+The [for-each loop](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html) provides a mechanism for executing a block of code for each element in a collection.
+Some documentation (eg. Oracle's documentation) refers to these as enhanced for loops.
```

Comment:
Link here for Oracle's documentation, maybe?

Reply:
The link in the previous sentence _is_ to Oracle's documentation so it seems redundant to double the link. The main reason I mention it is because Oracle's documentation does not call it a for-each loop.

Reply:
There you go.  I missed the link reference.  Since "some" is mentioned, is there a link where it does not do so?  Or should we say "The documentation" or "Oracle's documentation" instead?  Then it is an example, but it is explicitly riding on the essence of the link given?

Reply:
The markdown file we are commenting on (and exercism in general) would be the principal place where we use the terminology of for-each loops. I am sure there are more places, but have no need nor desire to track them down.

I think you may be right though, we could just say "Oracle's documentation" in order to achieve the same purpose I had in mind. I'll update the PR tomorrow to address that.